core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1009,8 +1006,8 @@ private DeleteResult deleteInLucene(Delete delete, DeletionStrategy plan)
         } catch (Exception ex) {
             if (indexWriter.getTragicException() == null) {
                 // there is no tragic event and such it must be a document level failure
-                return new DeleteResult(ex, plan.versionOfDeletion, plan.versionOfDeletion,
-                    plan.currentlyDeleted == false);
+                return new DeleteResult(
+                        ex, plan.versionOfDeletion, plan.seqNoOfDeletion, plan.currentlyDeleted == false);
```

Comment:
@areek I think that the tests in https://github.com/elastic/elasticsearch/pull/23314 don't cover the delete case (but only indexing). Can you look at this?